platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java

```diff
@@ -981,7 +981,14 @@ public MarkerView addMarker(@NonNull BaseMarkerViewOptions markerOptions) {
         long id = mapView.addMarker(marker);
         marker.setId(id);
         annotations.put(id, marker);
-        markerViewManager.invalidateViewMarkersInVisibleRegion();
+
```

Comment:
Correct, at time of patching I wasn't aware that this was expected behaviour as documented in #7072. I'm going to try invoking a render / waiting for a render to finish to solve this.